source/solvers/gls_navier_stokes.cc

```diff
@@ -1775,11 +1820,13 @@ GLSNavierStokesSolver<dim>::solve()
 
       if (this->simulation_control->is_at_start())
         {
+          this->define_dynamic_zero_constraints();
           this->iterate();
         }
       else
         {
           NavierStokesBase<dim, GlobalVectorType, IndexSet>::refine_mesh();
+          this->define_dynamic_zero_constraints();
           this->iterate();
         }
```

Comment:
I think this could be refactored to:
```suggestion
      if (!this->simulation_control->is_at_start())
          NavierStokesBase<dim, GlobalVectorType, IndexSet>::refine_mesh();

      this->define_dynamic_zero_constraints();
      this->iterate();
```